clients/go/compat.go

```diff
@@ -0,0 +1,275 @@
+package sapphire
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"sync"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/fxamacker/cbor/v2"
+	"github.com/twystd/tweetnacl-go/tweetnacl"
+)
+
+const (
+	DefaultGasPrice   = 100_000_000_000
+	DefaultGasLimit   = 30_000_000 // Default gas params are assigned in the web3 gateway.
+	DefaultBlockRange = 15
+)
+
+// packedTxs stores the txs that have been packed in the signer and therefore do not need to be packed in `backend.SendTransaction`.
```

Comment:
Is the only reason why this exists, to support `PackTx` without a `WrappedBackend` instance?

Reply:
[This](https://github.com/ethereum/go-ethereum/blob/e5eb32acee19cc9fca6a03b10283b7484246b15a/accounts/abi/bind/base.go#L392) is the problem. The transaction returned by `transact` and therefore any of the abigenerated methods is the one that comes out of the signer. If the signer is not wrapped, you'll get a very confusing output where the tx hashes and data are all wrong. The solution this library implements is to create a wrapped signer that creates the packed transaction. Unfortunately, you can use the wrapped signer independently of a `WrappedBackend` (not sure why, but you can) or a `WrappedBackend` independently of a special signer (quite likely if the user is uninformed). That means that both the signer _and_ `WrappedBackend` must be capable of packing unpacked txs. The problem is that the backend doesn't know that it's receiving an already-packed tx from the signer (unless it decrypts the tx, but that's costly and slightly more brittle). The compromise solution is to just store txs that have likely been packed and send those along without modification.